crates/vm/db.rs

```diff
@@ -53,10 +56,16 @@ impl revm::Database for StoreWrapper {
             .unwrap_or_else(|| RevmU256::ZERO))
     }
 
-    fn block_hash(&mut self, number: RevmU256) -> Result<RevmB256, Self::Error> {
-        self.store
-            .get_block_header(number.to())?
-            .map(|header| RevmB256::from_slice(&header.compute_block_hash().0))
-            .ok_or_else(|| StoreError::Custom(format!("Block {number} not found")))
+    fn block_hash(&mut self, number: u64) -> Result<RevmB256, Self::Error> {
+        let block_header = self
+            .store
+            .get_block_header(number)?
+            .ok_or_else(|| StoreError::Custom(format!("Block {number} not found")))?;
+
+        // Update the oldest_block_number
+        let mut oldest_block_number = self.oldest_block_number.borrow_mut();
+        *oldest_block_number = number.min(*oldest_block_number);
+
+        Ok(RevmB256::from_slice(&block_header.compute_block_hash().0))
```

Comment:
Is this behaviour intended? Where is it documented? What  is `oldest_block_number` used for?

Reply:
It's being used by the prover. [Here](https://github.com/lambdaclass/ethereum_rust/blob/981d952a6588bd255d5fbeea5b78791827ad7ade/crates/l2/operator/proof_data_provider.rs#L519), the PR is behind main by several commits, but that will be the use of `oldest_block_number`.

Reply:
removed for now